infrastructure/repository/contact_repository_test.go

```diff
@@ -0,0 +1,291 @@
+	assert.NoError(t, err)
+	assert.Equal(t, editedContact.UserID, updatedContact.UserID)
+	assert.Equal(t, editedContact.Username, updatedContact.Username)
+	assert.Equal(t, editedContact.FirstName, updatedContact.FirstName)
+	assert.Equal(t, editedContact.LastName, updatedContact.LastName)
+	assert.Equal(t, editedContact.PhoneNumber, updatedContact.PhoneNumber)
+}
+
+// TestDeleteContactUsername tests deleting a contact by username.
+func TestDeleteContactUsername(t *testing.T) {
+	repo, contact := setupContactTest(t)
+	defer cleanupContactTest(t, repo, contact)
+
+	err := repo.DeleteContactByUsername(*contact)
+	assert.NoError(t, err)
+
+	var deletedContact entities.PhoneBook
+	err = repo.DB.Where("user_id = ? AND username = ?", contact.UserID, contact.Username).First(&deletedContact).Error
+	assert.Error(t, err)
+}
+
+// TestGetContactList tests getting a list of contacts.
+func TestGetContactList(t *testing.T) {
+	repo, contact := setupContactTest(t)
+	defer cleanupContactTest(t, repo, contact)
+
+	// Contact with invalid UserID
+	contactWithInvalidUserID := entities.Contact{
+		UserID:      uuid.New(),
+		PhoneBookID: contact.PhoneBookID,
+	}
+	contacts, err := repo.GetContactList(contactWithInvalidUserID)
+	assert.NoError(t, err)
+	assert.NotNil(t, contacts)
+	assert.Empty(t, err)
+
+	// Contact with invalid PhoneBookID
+	contactWithInvalidPhoneBookID := entities.Contact{
+		UserID:      contact.UserID,
+		PhoneBookID: uuid.New(),
+	}
+	contacts, err = repo.GetContactList(contactWithInvalidPhoneBookID)
+	assert.NoError(t, err)
+	assert.NotNil(t, contacts)
+	assert.Empty(t, err)
+
+	// Contact with valid UserID and PhoneBookID
+	contacts, err = repo.GetContactList(*contact)
+	assert.NoError(t, err)
+	assert.NotNil(t, contacts)
+	assert.Len(t, contacts, 1)
+}
+
+// TestGetContactById tests getting a contact by ID.
+func TestGetContactById(t *testing.T) {
+	repo, contact := setupContactTest(t)
+	defer cleanupContactTest(t, repo, contact)
+
+	gContact, err := repo.GetContactById(*contact)
+	assert.NoError(t, err)
+	assert.Equal(t, contact.UserID, gContact.UserID)
+	assert.Equal(t, contact.PhoneBookID, gContact.PhoneBookID)
+	assert.Equal(t, contact.FirstName, gContact.FirstName)
+	assert.Equal(t, contact.LastName, gContact.LastName)
+	assert.Equal(t, contact.Username, gContact.Username)
+	assert.Equal(t, contact.PhoneNumber, gContact.PhoneNumber)
+
+	var contactWithInvalidId entities.Contact
+	contactWithInvalidId.ID = uuid.New()
+	contactWithInvalidId.UserID = contact.UserID
+	contactWithInvalidId.PhoneBookID = contact.PhoneBookID
+
+	_, err = repo.GetContactById(contactWithInvalidId)
+	assert.Error(t, err)
+
+	var contactWithInvalidUserId entities.Contact
+	contactWithInvalidUserId.ID = contact.ID
+	contactWithInvalidUserId.UserID = uuid.New()
+	contactWithInvalidUserId.PhoneBookID = contact.PhoneBookID
+
+	_, err = repo.GetContactList(contactWithInvalidUserId)
+	assert.Empty(t, err)
```

Comment:
it appears that these are testing functionality with Invalid entries, shouldn't they assert Error? if these are invalid, they SHOULD ERROR and NOT be empty/noerror